test/test_torch.py

```diff
@@ -17224,6 +17224,76 @@ def test_reciprocal_complex_extremal(self, device, dtype):
 
         self.compare_with_numpy(torch.reciprocal, np.reciprocal, vals, device, dtype)
 
+    def test_copysign_bool(self, device):
+        # Promoted to float
+        # Only 0 and 1
+        x = torch.tensor([0, 0, 1, 1], device=device, dtype=torch.bool)
+        y = torch.tensor([0, 1, 0, 1], device=device, dtype=torch.bool)
+        res = torch.tensor([0, 0, 1, 1], device=device, dtype=torch.float)
+        self.assertEqual(torch.copysign(x, y), res)
+
+    def _test_copysign(self, device, dtype, result_dtype):
+        # Scalar
+        x = torch.tensor([-1, 0, -0, 1], dtype=dtype)
+        y = torch.tensor(-1.)
+        res = torch.tensor([-1, -0, -0, -1], dtype=result_dtype)
+        self.assertEqual(torch.copysign(x, y), res)
+
+        x = torch.tensor(-1.)
+        y = torch.tensor([-1, 0, -0, 1], dtype=dtype)
+        res = torch.tensor([-1, 1, 1, 1], dtype=result_dtype)
+        self.assertEqual(torch.copysign(x, y), res)
+
+        # Constant
+        x = torch.tensor([-1, 0, -0, 1], dtype=dtype)
+        res = torch.tensor([-1, -0, -0, -1], dtype=result_dtype)
+        self.assertEqual(torch.copysign(x, -1.), res)
```

Comment:
We should also add a test for torch.copysign(x, 0.0) and torch.copysign(x, -0.0), just to cover the edge cases here